test/test_integration_credentials_netrc.py

```diff
@@ -0,0 +1,413 @@
+from test.util.test_infrastructure import integration_test
+from test.util.test_runners.integration_test_case import IntegrationTestCaseBase
+from test.util import test_helpers as utils
+
+from conjur.constants import DEFAULT_NETRC_FILE, DEFAULT_CONFIG_FILE, DEFAULT_CERTIFICATE_FILE
+
+@patch('conjur.wrapper.keystore_adapter.KeystoreAdapter.is_keyring_accessible', return_value=False)
+class CliIntegrationTestCredentialsNetrc(IntegrationTestCaseBase):
+    # *************** HELPERS ***************
+    def __init__(self, testname, client_params=None, environment_params=None):
+        super(CliIntegrationTestCredentialsNetrc, self).__init__(testname, client_params, environment_params)
+
+
+    def setup_cli_params(self, env_vars, *params):
+        self.cli_auth_params = ['--debug']
+        self.cli_auth_params += params
+
+        return self.cli_auth_params
+
+    def setUp(self):
+        self.setup_cli_params({})
+        try:
+            utils.remove_file(DEFAULT_NETRC_FILE)
+            utils.remove_file(DEFAULT_CONFIG_FILE)
+            utils.remove_file(DEFAULT_CERTIFICATE_FILE)
+        except OSError:
+            pass
+
+    def validate_netrc(self, machine, login, password):
+        with open(DEFAULT_NETRC_FILE, 'r') as netrc:
+            lines = netrc.readlines()
+            assert f"machine {machine}" in lines[0]
+            assert f"login {login}" in lines[1]
+            assert f"password {password}" in lines[2]
+
+    def write_to_netrc(self, machine, login, password):
+        with open(f"{DEFAULT_NETRC_FILE}", "w") as netrc_test:
+            netrc_test.write(f"machine {machine}\n")
+            netrc_test.write(f"login {login}\n")
+            netrc_test.write(f"password {password}\n")
+
+    # *************** LOGIN CREDENTIALS TESTS ***************
+
+    '''
+    Validates that when a user already logged in and reattempts and fails, the previous successful session is not removed
+    '''
+
+    @integration_test(True)
+    def test_https_netrc_was_not_overwritten_when_login_failed_but_already_logged_in(self,mock_accessible):
+        utils.setup_cli(self)
+        successful_run = self.invoke_cli(self.cli_auth_params,
+                                         ['login', '-i', 'admin', '-p', self.client_params.env_api_key])
+        self.assertIn("Successfully logged in to Conjur", successful_run.strip())
+
+        unsuccessful_run = self.invoke_cli(self.cli_auth_params,
+                                           ['login', '-i', 'someinvaliduser', '-p', 'somewrongpassword'], exit_code=1)
+        self.assertRegex(unsuccessful_run, "Reason: 401 Client Error: Unauthorized for")
+        self.validate_netrc(f"{self.client_params.hostname}", "admin", self.client_params.env_api_key)
+
+    '''
+    Validates when a user can logout successfully
+    '''
+
+    @integration_test(True)
+    def test_https_logout_successful_netrc(self,mock_accessible):
```

Comment:
please add why we need these tests twice. I understand because of mock_accessible which forces us to use netrc but this should be in the comment above

Reply:
because it is easily missed